lib/restful_resource/http_client.rb

```diff
@@ -234,8 +240,8 @@ def build_user_agent(app_name)
 
     def http_request(request)
       response = @connection.send(request.method) do |req|
-        req.options.open_timeout = request.open_timeout || default_open_timeout # seconds
-        req.options.timeout = request.timeout || default_timeout # seconds
+        req.options.timeout = request.timeout if request.timeout
```

Comment:
won't the timeout always be set now? is there a scenario when we wouldn't want to assign anything here if it's not set?